### PR TITLE
Get "make dist" working properly, including build and clean

### PR DESCRIPTION
**Pull request by kgaillot**
Our CI infrastructure will soon require that "make distcheck" work. This is the first major step toward that, getting distcheck's first step working: "make dist" followed by a VPATH build in the resulting distribution. (As of this commit, the resulting build is not yet usable, and "make check" fails.)

This required an overhaul of many Makefiles, most significantly to ensure that all the right files are distributed, and that rules can handle the build directory being separate from the source directory.

A good bit of clean-up and reorganization was done as well, to fix minor issues and make the logic clearer. This included improving our usage of silent rules.

### CODE REVIEW

**Comment by kgaillot** [comment]
@jnpkrn , can you review this? The goal is basically that "make dist" produces a distribution that "./configure && make && make clean" works in, and that has all the files an end user might reasonably want.

**Comment by kgaillot** [comment]
rebased on current master to resolve conflict, otherwise unchanged

File: doc/Makefile.am [hunk 1]
```diff
@@ -93,12 +93,13 @@ publican_docs	+= $(docbook)
 endif
 endif
 
-EXTRA_DIST	= $(ascii) $(SHARED_TXT) $(PNGS_ORIGINAL) $(SVGS)
+EXTRA_DIST	= $(ascii) $(SHARED_TXT) $(PNGS_ORIGINAL) $(DOTS) $(SVGS)
```
Comment (jnpkrn) [inline doc/Makefile.am]: What's the point of shipping dot files when they are only ever the
intermediate byproducts in the documentation context and given that
the derived files are (likely) shipped as well?

Distribution tarballs don't aim to prove the whole chain of
intermediaries till the final artefacts are reached.

Comment (jnpkrn) [reply doc/Makefile.am]: Oh, I see the context now, dots are not generated, there's a TODO
for that.

**Comment by jnpkrn** [comment]
Aside wrt. `Build: GNUmakefile: allow all/clean to work without git `:
I think that's a heavy concentration of some git magic to solve something
there are stock solutions for, i.e., the move in libqb direction for instance
would make sense long-term.  It would also have a greater expressiveness
than `export-subst` in `.gitattributes` has, IIRC.  And it solves the problem
of not-in-checkout pretty well.

File: doc/Makefile.am [hunk 2]
```diff
@@ -28,22 +45,13 @@ RSYNC_OPTS      = -rlptvzxS --progress
 LAST_RELEASE	?= Pacemaker-$(VERSION)
 TAG		?= $(shell git log --pretty=format:%H -n 1 HEAD)
 
-publican_docs   =
```
Comment (jnpkrn) [inline doc/Makefile.am]: ah, in fact, was never used :)

File: doc/Makefile.am [hunk 3]
```diff
@@ -138,144 +130,129 @@ COMMON_XML = Author_Group.xml Book_Info.xml Revision_History.xml
 SHARED_TXT=$(wildcard shared/en-US/*.txt)
 SHARED_XML=$(SHARED_TXT:%.txt=%.xml)
 
+if PUBLICAN_INTREE_BRAND
+PUBLICAN_INTREE_DEPS	= publican-catalog
+PUBLICAN_INTREE_ENV	= XML_CATALOG_FILES="$(CURDIR)/publican-catalog"
+PUBLICAN_INTREE_OPT	= --brand_dir=../publican-clusterlabs
```
Comment (jnpkrn) [inline doc/Makefile.am]: trailing whitespace above

File: doc/Makefile.am [hunk 4]
```diff
@@ -392,12 +369,12 @@ if BUILD_DOCBOOK
 		rsync $(RSYNC_OPTS) $$book/publish/* "$(RSYNC_DEST)/$(PACKAGE)/doc/";			\
 	done
 endif
-	rsync $(RSYNC_OPTS) $(generated_docs) $(ascii) "$(RSYNC_DEST)/$(PACKAGE)/doc/"
+	rsync $(RSYNC_OPTS) $(doc_DATA) "$(RSYNC_DEST)/$(PACKAGE)/doc/"
 
 
 clean-local:
 	-rm -f $(PNGS_GENERATED)
 	-rm -rf $(generated_docs) $(generated_mans) $(docbook_build)
```
Comment (jnpkrn) [inline doc/Makefile.am]: the above still refers to `generated_mans`

**Comment by kgaillot** [comment]
Thanks!

**Comment by jnpkrn** [comment]
Also, I think that merging intentionally split lightweight and
build-machinery-free checks where possible (giving feedback
in tens of seconds rather than in order or two of magnitude
later) is a step backward.

**Comment by jnpkrn** [comment]
Also there is practically unenumerable amount of other plain code-as-is
checks to possibly add to what was `MAINT=1` originally meant to be.
(it was intentionally open-ended for when there's time to append more
on-case-encountered checks).

For instances, I don't think correctly configured git client is
currently a subject of manual review, hence it would be trivial to
programmatically look at whether Author in commit metadata doesn't,
by any chance, contain something other than valid looking e-mail
address.  There's certainly a "market" for fully isolated `MAINT=1`
CI run.

**Comment by kgaillot** [comment]
I was hesitant, but we need VPATH builds to work, and that means we need builddir etc. automake variables produced after autogen and configure.

I avoided a performance penalty by still only doing the schema tests once (with gcc and not clang).

**Pull request merged**